### PR TITLE
Introduce `GraphqlContext` type

### PR DESCRIPTION
**Pull request by simonvandel**
Will make calling internal nicer since we don't need to construct a random json value not knowing what the possibilities are.

Blocked & based on https://github.com/BlackbirdHQ/bb_rust/pull/18 which should be reviewed first.

### CODE REVIEW

File: src/graphql/internal.rs [hunk 1]
```diff
@@ -67,3 +91,97 @@ struct InternalGraphQLResponse<T> {
...
+            default_language: Default::default(),
+            language: Default::default(),
+            group_ids: Default::default(),
+            line_ids: Default::default(),
+            peripheral_ids: Default::default(),
+            user_pools: Default::default(),
+            user_sub: Default::default(),
+            user_pool,
+            required_by: Default::default(),
+            requires: Default::default(),
```
Comment (MathiasKoch) [inline src/graphql/internal.rs:139]:
```suggestion
            user_pool,
            ..Default::default()
```

File: src/graphql/internal.rs [hunk 2]
```diff
@@ -67,3 +91,97 @@ struct InternalGraphQLResponse<T> {
     data: Option<T>,
     errors: Option<serde_json::Value>,
 }
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
```
Comment (MathiasKoch) [inline src/graphql/internal.rs:95]:
```suggestion
#[derive(Default, Deserialize, Serialize, Debug, Clone)]
```

**Review by MathiasKoch — APPROVED** [review]

File: src/graphql/internal.rs (hunk 2 above)
Comment (simonvandel) [reply src/graphql/internal.rs:95]: I intentionally did not derive Default, as I wanted the user to always put in a user pool in the `new` function. What are your thoughts on that?

Comment (MathiasKoch) [reply src/graphql/internal.rs:95]: Yeah, i kinda figured that much which is also why the suggestion was part of "approved" :p It makes sense as long as it is reasoned.

**Pull request merged**